### ISSUE TITLE
should have a function to read in IOOS NGDAC netcdf format files

### ISSUE THREAD

**Issue by clayton33** [dankelley/oceanglider#93]
Effort has been made to put SeaExplorer and Slocum glider data collected by DFO and affiliated partners into the IOOS NGDAC netcdf format using the python glider processing chain, `pyglider`. I think it might be a good idea to put some effort into reading in this file type so data can be used and visualized by `R` savvy users.

Proposed function name : `read.glider.netcdf.ioos()`

Documentation on format : https://ioos.github.io/ioosngdac/ngdac-netcdf-file-format-version-2.html

**Comment by dankelley** [comment]
It would be great to have a sample file, either in the repo (if it is very small) or otherwise.  Netcdf is not hard to read so I think the main task will be deciding on how to store the data (data frames? lists? matrices?) and how to name the variables (use oce names, but with 'originalNames' as in the data file).

**Comment by clayton33** [comment]
A sample file from data that I've processed is about 5mb, is that too big ? If not, where should I put it ?

Also, I'm planning on downloading a file from the IOOS ERDDAP site... but it might have to wait until this afternoon, I think I'm blocked from downloading data through R on work's network.

**Comment by dankelley** [comment]
I'd say email the file to me and Clark, as the other main oce authors.  That way all 3 of us will have it.  I think all 3 of us are quite familiar with reading netcdf files, and so this is only a matter of (a) deciding how to structure the data so that it fits in this package and (b) figuring out how to map the netcdf varnames to the varnames used in the package.

I'm curious and if you email me the file, most likely I'll look at it tomorrow (quite busy today)

**Comment by clayton33** [comment]
Roger that. I'll send them along shortly.

**Comment by dankelley** [comment]
Some code to read/plot:
```R
library(oce)
library(ncdf4)
f <- "ga_538-20151124T1730-delayed.nc"
o <- nc_open(f)
# See what we have
varNames <- names(o$var)
print(varNames)

# Extract a few standard things to see if data look okay
# use as.vector because netcdf stores as a 1D array thingee-doodle
pressure <- as.vector(ncvar_get(o, "pressure"))
salinity <- as.vector(ncvar_get(o, "salinity"))
temperature <- as.vector(ncvar_get(o, "temperature"))
latitude <- as.vector(ncvar_get(o, "latitude"))
longitude <- as.vector(ncvar_get(o, "longitude"))

time <- numberAsPOSIXct(as.vector(ncvar_get(o, "time")), tz="UTC")
preciseTime <- numberAsPOSIXct(as.vector(ncvar_get(o, "precise_time")), tz="UTC")
ctd <- as.ctd(salinity, temperature, pressure, latitude=latitude, longitude=longitude)
# There are some very low S values so we chop them
eps <- 0.001
Slim <- quantile(salinity, c(eps, 1-eps), na.rm=TRUE)
Tlim <- quantile(temperature, c(eps, 1-eps), na.rm=TRUE)
plotTS(ctd, eos="unesco", Slim=Slim, Tlim=Tlim, cex=0.3)
plot(longitude, latitude, asp=1/cos(pi/180*mean(range(latitude, na.rm=TRUE))))
plot(pressure) # sequence plot
# 'time' has just 2 values (one per swoop, I suppose) ...
table(time)
# ... but 'precise_time' has values mapped to pressures
oce.plot.ts(preciseTime, pressure, type="o")
```

**Comment by clayton33** [comment]
I have this working in my branch, `cl`,  to a point that I'm somewhat happy with it, but I'll still be doing some development as things progress. We'll still have to figure out how to deal with `flag` and `qc` variables, but I'll leave that for another time. I'll close this, and we can open other issues for this function, as needed. 

@dankelley should we merge my branch into `develop` or leave it for now ?

**Comment by dankelley** [comment]
@clayton33 sorry, I didn't see your Q about merging into `develop`. I've just merged it into `develop`, in commit 588af21e5e8a1b38ad579e07597398444034b56e.  I think dealing with flags and QC variables ought to be another issue or issues.

I'm not too sure on whether flags and QC are different things, which is why I say 'issue or issues'.  Anyway, a good first step in figuring out what to do with flags/qc might be to find a file that contains such data.  I tried as below but that file has only NA values for the variables I've checked.
```R
> o<-ncdf4::nc_open("sandbox/cl/01downloadFromIOOSERDDAP/ga_538-20151124T1730-delayed.nc")
> any(!is.na(ncdf4::ncvar_get(o,"salinity_qc")))
[1] FALSE
```
